Add defaults to swaplevel() parameters i and j

brandon-rhodes opened the pull request:
Closes #12934 feature request.

brandon-rhodes committed:
Add default values to swaplevel() i, j parameters

jreback commented:
@brandon-rhodes I just updated master, so you will need to rebase as unrelated issue with python-dateutil will cause a test for fail :< (we are now skipping)

jreback commented on doc/source/whatsnew/v0.18.1.txt:
```diff
@@ -131,6 +131,8 @@ These changes conform sparse handling to return the correct types and work to ma
 
 API changes
 ~~~~~~~~~~~
+- ``.swaplevel()`` for ``Series``, ``DataFrame``, ``Panel``, and ``MultiIndex`` now features defaults for its first two parameters ``i`` and ``j`` that swap the two innermost levels of the index.
```
add the issue number at the end, don't add a blank line

brandon-rhodes replied:
To make sure I understand: are you saying that (a) even though the subsequent bullet points are separated by blank lines, the blank line beneath this bullet point should disappear? or (b) that when I go in to the code to add the issue number, I should keep it on the same line and not let it land by accident on a subsequent line?

jreback replied:
oh I just meant no need to add the line in between. its doesn't matter much as sphinx will ignore it, but just trying to be consistent (though I think there are some extra lines there anyhow....)

jreback commented:
did u mean to close this?

brandon-rhodes commented:
No. I am used to the Phabricator work flow, and this pull request is teaching me the GitHub workflow :) Stand by!

brandon-rhodes commented:
@jreback Okay, I pulled and pushed some stuff, and all the code reappeared. :) I added the issue number, and rebased against your latest master — let me know how things look!

jreback commented:
can u update tests for MultiIndex as well

jreback commented:
pls squash as well

jreback commented on pandas/core/frame.py:
```diff
@@ -3384,7 +3384,7 @@ def nsmallest(self, n, columns, keep='first'):
-    def swaplevel(self, i, j, axis=0):
+    def swaplevel(self, i=-2, j=-1, axis=0):
         """
         Swap levels i and j in a MultiIndex on a particular axis
```
I think it's worth noting in the doc strings (of alll of these) that the defaults were added in 0.18.1

you can use a versionadded directive

brandon-rhodes commented:
Oh — GitHub doesn't let you squash as you land the pull request? Yet another difference with the workflow I'm used to. Yes, I'll do the squashing myself, and also try addressing your two comments — thanks!

jreback commented:
yes I will squash on merge - since we rebase its usually a bit easier to use logical commits and squash

brandon-rhodes commented:
Just saw your comment — had already squashed on my end :) Per the Sphinx docs, I went with ``versionchanged`` instead, since it seemed closer to the intent of a change like this. I also found the only test I could find where the multiindex tests used swaplevel and supplemented it. Let me know what you think! (Also: wow, your integration tests take a while!)

jreback commented:
@brandon-rhodes lgtm.

any comments? @sinhrks @shoyer @jorisvandenbossche @TomAugspurger

shoyer commented:
This seems reasonable enough to me. :+1:

TomAugspurger commented:
👍

jreback commented:
thanks @brandon-rhodes

brandon-rhodes commented:
Thanks for considering this contribution from me! :)
